src/EVM/ABI.hs

```diff
@@ -516,6 +533,7 @@ parseAbiValue (AbiArrayType n typ) =
   AbiArray n typ <$> do a <- listP (parseAbiValue typ)
                         return $ Vector.fromList a
 parseAbiValue (AbiTupleType _) = error "tuple types not supported"
+parseAbiValue AbiFunctionType = error "function types not supported"
```

Comment:
Is there a reason for this? I puls prefer to have it implemented unless it's extremely inconvenient

Reply:
As far as I know, the only ways to make a function pointer constant in solidity are putting the name of a function (we couldn't turn that into its raw bytes without more context, so we couldn't construct a `AbiFunction` value) or making a bytes24 with the value you want and then using `assembly { a := b }` to skip type checking and assign the function pointer to be the bytes24 value (kind of a hack; if you want, through, we could make `parseAbiValue AbiFunctionType` parse bytes24)

Reply:
yes I think that makes sense, we use this code when constructing partially symbolic calldata buffers for symbolic execution (we parse a user provided string into some abi value and write that as a concrete region over a symbolic buffer), and I think that it would be nice to be able to do this for function pointers as well.

Reply:
Okay, done